pkg/pod/entrypoint_lookup_impl.go

```diff
@@ -56,18 +57,24 @@ func NewEntrypointCache(kubeclient kubernetes.Interface) (EntrypointCache, error
 // It also returns the digest associated with the given reference. If the
 // reference referred to an index, the returned digest will be the index's
 // digest, not any platform-specific image contained by the index.
-func (e *entrypointCache) get(ctx context.Context, ref name.Reference, namespace, serviceAccountName string) (*imageData, error) {
+func (e *entrypointCache) get(ctx context.Context, ref name.Reference, namespace, serviceAccountName string, imagePullSecrets []corev1.LocalObjectReference) (*imageData, error) {
 	// If image is specified by digest, check the local cache.
 	if digest, ok := ref.(name.Digest); ok {
 		if id, ok := e.lru.Get(digest.String()); ok {
 			return id.(*imageData), nil
 		}
 	}
 
+	var pullSecretsNames []string
```

Comment:
nit: We already know the size of this list, so we can just initialize it with the set capacity.

```suggestion
	pullSecretsNames := make([]string, 0, len(imagePullSecrets))
```